.github/workflows/test-phpunit.yml

```diff
@@ -165,33 +169,79 @@ jobs:
         uses: actions/github-script@v6
         id: phpunit-coverage-option
         with:
-          script: 'return "${{ matrix.php-versions }}" == "8.0" ? "" : "--no-coverage"'
+          script: |
+            const { COVERAGE_NAME } = process.env
+
+            return "${{ matrix.php-versions }}" == "8.0" ? `--coverage-php build/cov/coverage-${COVERAGE_NAME}.cov` : "--no-coverage"
           result-encoding: string
+        env:
+          COVERAGE_NAME: php-v${{ matrix.php-versions }}-${{ matrix.db-platforms }}
 
       - name: Test with PHPUnit
         run: script -e -c "vendor/bin/phpunit --color=always --exclude-group=auto-review ${{ steps.phpunit-coverage-option.outputs.result }}"
         env:
           DB: ${{ matrix.db-platforms }}
           TERM: xterm-256color
 
-      - name: Run Coveralls
-        if: github.repository_owner == 'codeigniter4' && matrix.php-versions == '8.0'
-        run: |
-          composer global require --ansi php-coveralls/php-coveralls:^2.4
-          php-coveralls --coverage_clover=build/logs/clover.xml -v
+      - name: Upload coverage file
+        if: matrix.php-versions == '8.0'
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.COVERAGE_NAME }}
+          path: build/cov/coverage-${{ env.COVERAGE_NAME }}.cov
+          if-no-files-found: error
+          retention-days: 5
         env:
-          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          COVERALLS_PARALLEL: true
-          COVERALLS_FLAG_NAME: PHP ${{ matrix.php-versions }} - ${{ matrix.db-platforms }}
+          COVERAGE_NAME: php-v${{ matrix.php-versions }}-${{ matrix.db-platforms }}
 
-  coveralls-finish:
+  coveralls:
     if: github.repository_owner == 'codeigniter4'
-    needs: [tests]
+    needs: tests
     runs-on: ubuntu-20.04
 
     steps:
-      - name: Coveralls Finished
-        uses: coverallsapp/github-action@master
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup PHP, with composer and extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ env.COVERAGE_PHP_VERSION }}
+          tools: composer
+          coverage: xdebug
```

Comment:
We aren't actually running anything in this step, right? I think this is unnecessary.

Reply:
Which one? the php setup or the coverage driver?

Reply:
Line 209. Basically: do we need Xdebug for this operation? It seems unlikely, and having debugging on often slows down operations.

Reply:
It seems xdebug is needed by phpcov.
https://github.com/codeigniter4/CodeIgniter4/actions/runs/3306905998/jobs/5458244739

Reply:
Well darn.